pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -847,18 +847,21 @@ func addAddresses(secrets corecontrollers.SecretCache, config map[string]interfa
 		return
 	}
 
-	if machine.Annotations["objectset.rio.cattle.io/id"] == "unmanaged-machine" || machine.Spec.InfrastructureRef.Kind == "PodMachine" {
+	if config["cloud-provider-name"] != nil || machine.Annotations["objectset.rio.cattle.io/id"] == "unmanaged-machine" {
 		// Unmanaged machines won't have a state secret,
-		// and PodMachine's IP address is not external
+		// and if the cloud provider will handle this, if it is set.
```

Comment:
```suggestion
		// and the cloud provider will handle this, if it is set.
```
Super-nit, but I think first `if` is not needed in this comment.

Reply:
Done.